pandas/core/generic.py

```diff
@@ -8564,6 +8523,179 @@ def nanptp(values, axis=0, skipna=True):
             ``numpy.ndarray`` method ``ptp``.""",
             nanptp)
 
+    _shared_docs['transform'] = ("""
+    Apply a function that returns a same-shaped object.
+
+    .. versionadded:: 0.20.0
+
+    Parameters
+    ----------
+    func : callable, list of callables, or dict
+        The function or function to apply to the DataFrame or Series.
+        Each function should expect a Series and return an array-like
+        with the same shape. Accepted combinations are:
+
+        - function : Applied to the Series, or to each column of the
+          DataFrame independently.
+        - list of functions : Each function is applied to the Series
+          or to each column of the DataFrame independently.
+        - dict of column names -> functions or list of functions :
+          For DataFrames only, apply the function or functions for a
+          given key to a specific column.
+
+    axis : {0 or 'index', 1 or 'columns'}, default 'index'
+        Axis to apply the function along (DataFrame only).
+
+        * 0 or 'index' : apply along the index, i.e. the function
+          gets each column.
+        * 1 or 'columns' : apply along the columns, i.e. the function
+          gets each row.
+
+    *args
+        Additional positional arguments are passed to `func`.
+    **kwargs
+        Additional keyword arguments are passed to `func`.
```

Comment:
You can merge both in a single line.

Reply:
This PR is only about documentation. No change whatsoever in code. It was initiated during the sprint - and I was wondering why it was left hanging..

Done: I merged both lines into a single one.

Reply:
There are code changes, mainly replacing lambdas by named functions. May be your IDE did it for you?

Reply:
Argh... Yes I'm afraid. Sorry about that. Unintended.